### ISSUE TITLE
Multiline labels which start or end on the same line should have their lines merged

### ISSUE THREAD

**Issue by RDambrosio016** [brendanzab/codespan#265]
Currently if you have two multiline labels, which both start or end on the same line, the rendering gets kind of wonky 
```
error[no-empty]: Empty block statements are not allowed
  ┌─ tests\main.js:1:5
  │
1 │     try {
  │ ┌───────^
2 │ │   
3 │ │   } catch (e {
  │ └───^
  │   ┌───'
4 │   │ 
5 │   │ } finally {
  │   └─' This handler is unreachable
```
Instead, the lines should be merged into one line as follows, with the lines after the first label being the color of the second label, in this case cyan, this would illustrate more clearly that the label does in fact start there
```
error[no-empty]: Empty block statements are not allowed
  ┌─ tests\main.js:1:5
  │
1 │     try {
  │ ┌───────^
2 │ │   
3 │ │   } catch (e {
  │ └─┬─^─'
4 │   │ 
5 │   │ } finally {
  │   └─' This handler is unreachable
```

**Comment by brendanzab** [comment]
Would this be ok?
```
error[no-empty]: Empty block statements are not allowed
  ┌─ tests\main.js:1:5
  │
1 │     try {
  │ ╭───────^
2 │ │   
3 │ │   } catch (e {
  │ ╰───^─'
4 │   │ 
5 │   │ } finally {
  │   ╰─' This handler is unreachable
```

**Comment by RDambrosio016** [comment]
Yeah i realize that you cant have them be the same color, however, i was thinking of two possible solutions:
- If the colors for primary and secondary are default, and this isnt a windows console you could blend them into purple using termcolor Rgb(..) which i think would look nice
- Or else just make the color of the junction the color of the first line, so red because of primary

I think the way you proposed looks sort of disconnected, but thats just my opinion 🙂

**Comment by brendanzab** [comment]
Yeah I'm really not a fan of either of those options TBH. In the first case you'd have a single character that is a different colour from the rest. In the second you'd have a distracting 'spur' of the primary label colour bleeding into the secondary underline. It does look disconnected, but it will look a bit better when there is some colour linking the two bits of the secondary line.

**Comment by RDambrosio016** [comment]
Yeah i understand, on second thought your proposal sounds better. its disconnected, however, thinking about it, a weird coloring for the bottom bit looks more frustrating and ugly 👍

**Comment by Johann150** [comment]
I think this should stay how it is. The proposal would make the rendering depend on colour output. If there is uncoloured output (or accessibility: colourblind) it looks really confusing, as can be seen in the examples given here.

Still, in the given examples one could differentiate between the labels because the first is a primary and the other is a secondary label and they thus use different characters for rendering (primary uses `^` and secondary uses `'`). But if both were primary or secondary labels, it could easily look like the following, which is just more confusing:

```
error[no-empty]: Empty block statements are not allowed
  ┌─ tests\main.js:1:5
  │
1 │     try {
  │ ╭───────^
2 │ │
3 │ │   } catch (e {
  │ ╰───^─^
4 │   │
5 │   │ } finally {
  │   ╰─^ This handler is unreachable
```

Although I have to admit that `^─^` looks nice, as a whole this looks more like a downgrade to me.
